tls/s2n_ktls_io.c

```diff
@@ -271,3 +271,54 @@ S2N_RESULT s2n_ktls_recvmsg(struct s2n_connection *conn, uint8_t *record_type, u
     *bytes_read = result;
     return S2N_RESULT_OK;
 }
+
+static S2N_RESULT s2n_ktls_new_iovecs_with_offset(const struct iovec *bufs,
+        size_t count, size_t offs, struct s2n_blob *mem)
+{
+    RESULT_ENSURE(bufs != NULL || count == 0, S2N_ERR_NULL);
+    RESULT_ENSURE_REF(mem);
+
+    RESULT_GUARD_POSIX(s2n_realloc(mem, sizeof(struct iovec) * count));
```

Comment:
Who is responsible for freeing this memory?

Reply:
This DEFER_CLEANUP is: https://github.com/aws/s2n-tls/blob/0c4f993fa0962dff79ad98cce1cd52bd832ec6aa/tls/s2n_ktls_io.c#L311 As noted in https://github.com/aws/s2n-tls/pull/4174#discussion_r1310671802, the memory never leaves our library.